src/util/debug.c

```diff
@@ -287,13 +292,34 @@ void sss_vdebug_fn(const char *file,
          * searchable.
          */
         va_copy(ap_fallback, ap);
-        ret = journal_send(file, line, function, level, format, ap);
+        if (debug_chain_id > 0) {
+            result_fmt = chain_id_fmt_fixed;
+            ret = snprintf(chain_id_fmt_fixed, sizeof(chain_id_fmt_fixed),
+                           DEBUG_CHAIN_ID_FMT"%s", debug_chain_id, format);
+            if (ret < 0) {
+                return;
+            } else if (ret > sizeof(chain_id_fmt_fixed)) {
+                ret = asprintf(&chain_id_fmt_dyn, DEBUG_CHAIN_ID_FMT"%s",
+                               debug_chain_id, format);
+                if (ret < 0) {
+                    return;
+                }
+                result_fmt = chain_id_fmt_dyn;
+            }
+
+            ret = journal_send(file, line, function, level, result_fmt, ap);
+        } else {
+            ret = journal_send(file, line, function, level, format, ap);
+        }
         if (ret != EOK) {
             /* Emergency fallback, send to STDERR */
             vfprintf(stderr, format, ap_fallback);
             fflush(stderr);
         }
         va_end(ap_fallback);
+        if (chain_id_fmt_dyn != NULL) {
+            free(chain_id_fmt_dyn);
```

Comment:
It's safe to supply NULL to `free()`, man page: `If ptr is NULL, no operation is performed.`

Reply:
And I would move it right after `journal_send(file, line, function, level, result_fmt, ap);` but that's matter of taste, of course.

Reply:
Thanks, fixed.